src/hooks/useInputTokenBalance.ts

```diff
@@ -8,21 +8,22 @@ import { multiplyByPowerOfTen } from '../helpers/contracts';
 import { nativeToDecimal, USDC_DECIMALS } from '../helpers/parseNumbers';
 import { usePolkadotWalletState } from '../contexts/polkadotWallet';
 import { useAssetHubNode } from '../contexts/polkadotNode';
+import { useVortexAccount } from './useVortexAccount';
 
+// TODO maybe improve: if the user switches the network in the selector and REJECTS the switch in wallet, then balance will be 0.
```

Comment:
Metamask needs this explicit acceptance, while Talisman for instance does not.

Since `useReadContract` uses the chain that is connected in the wallet, it will read form an incorrect chain.
If this is a UI requirement, we could fetch with the correct chain using the `selectedNetwork` parameter.